templates/vue/src/components/Lightbox/media/VideoMedia/H5PVideoMedia/H5PIframe.vue

```diff
@@ -232,21 +253,51 @@ export default {
       }
     },
     handlePlay() {
-      this.$emit("show-play-screen", false)
-      if (!this.playedOnce && this.autoplay) {
-        this.playedOnce = true
-        return
-      }
       const { id, progress, mediaDuration } = this.node
       client.recordAnalyticsEvent("user", "play", "h5p-video", id, {
         time: progress * mediaDuration,
       })
+      this.$emit("play")
     },
     handlePause() {
-      this.$emit("show-play-screen", true)
-      const { id, progress, mediaDuration } = this.node
-      client.recordAnalyticsEvent("user", "pause", "h5p-video", id, {
-        time: progress * mediaDuration,
+      const video = this.getInstance()
+      if (video) {
+        /**
+         * When an H5PInteractiveVideo ends, it emits a Pause event BEFORE an Ended
+         * event. This breaks our state machine since our machine doesn't allow
+         * transitioning from a Paused state to a Finished state. As a work around,
+         * we listen to the Pause event and manually check if the video is done at
+         * this point. If it is, we emit the corresponding `timeupdate` event.
+         */
+        if (video.getCurrentTime() === video.getDuration()) {
+          this.$emit("timeupdate", {
+            amountViewed: 1,
+            currentTime: video.getDuration(),
+          })
+        } else {
+          const { id, progress, mediaDuration } = this.node
+          client.recordAnalyticsEvent("user", "pause", "h5p-video", id, {
+            time: progress * mediaDuration,
+          })
+          let hasDialogue = this.$refs.h5p.contentWindow.H5P.$body[0].querySelector(
+            ".h5p-dialog-wrapper"
+          )
+          if (
+            hasDialogue.style.display === "none" ||
+            hasDialogue.style.display === ""
+          ) {
+            this.$emit("pause")
+          }
+        }
+      }
+    },
+    handleVideoLoad(currentTime) {
+      this.$emit("load", {
+        type: "h5p-video",
+        currentTime,
+      })
+      this.$nextTick(() => {
+        this.setFrameDimensions()
       })
     },
     handleLoad() {
```

Comment:
There is some code here that assumes h5p might not be an interactive video (E.g. `if (this.library !== "H5P.InteractiveVideo")`, `h5pInstance.recorder`, `switch (this.library)`) these all should be removed since this component is only called for H5P interactive videos.